### PR TITLE
Move all SYCL headers into SYCL directory

### PR DESCRIPTION
**Pull request by masterleinad**
Cleaning up `core/src`.

### CODE REVIEW

File: core/src/SYCL/Kokkos_SYCL.cpp
Comment (ldh4) [inline core/src/SYCL/Kokkos_SYCL.cpp:23]: It looks like there are currently some inconsistencies between where each backend's `Kokkos_[backend].hpp` is placed in.
`Cuda`, `HPX`, `OpenMPTarget` and `Threads` have their `Kokkos_[backend].hpp` files still in `core/src` whereas some of others are in their respective directory.
Is there a reason why `Kokkos_SYCL.hpp` was placed in `core/src` initially but is being moved now?

**Review by ldh4 — APPROVED** [review]

**Review by dalg24 — APPROVED** [review]

File: core/src/SYCL/Kokkos_SYCL.cpp
Comment (dalg24) [reply core/src/SYCL/Kokkos_SYCL.cpp:23]: You are right, we are inconsistent at the moment.
What we want is what is implemented in this PR.
HPX and OpenMPTarget have ongoing open refactor PRs which is the main reason why we decided not to do one big PR but defer to individual backend maintainers.

**Pull request merged**
